packages/access/lib/constraints/limitGrants.js

```diff
@@ -0,0 +1,49 @@
+const debug = require('debug')('access:lib:constraints:limitGrants')
+
+/**
+ * Constraint limits grants.
+ *
+ * Story: There is a limited amount of grants a user can share per campaign.
+ *
+ * @example: {"limitGrants": {"grants": 2}}
+ */
+
+const isGrantable = async (args, context) => {
+  const { settings, granter, campaign } = args
+  const { pgdb } = context
+
+  const grants = await pgdb.query(`
+    SELECT "accessGrants".id
+
+    FROM "accessGrants"
+
+    WHERE
+      "accessGrants"."accessCampaignId" = '${campaign.id}'
+      AND "accessGrants"."granterUserId" = '${granter.id}'
```

Comment:
Would like to stick to either pattern in all constraint files.

[requirePackage.js](https://github.com/orbiting/backends/blob/feat-preview-with-access-grants/packages/access/lib/constraints/requirePackage.js#L17-L35) has a JOIN query, using template strings.

Reply:
You should not make the exception the norm. It's wrong in requirePackage.js as well. And dangerous, as it's not visible where `settings` is coming from. And at some point there might be user generated data in it, opening the door for SQL-injection. If I understand [packages](https://github.com/orbiting/backends/blob/feat-preview-with-access-grants/packages/access/lib/constraints/requirePackage.js#L15) correctly, you build an array of strings. This can be done with pogis variable interpolation easily, see for example [Notifications.js](https://github.com/orbiting/backends/blob/feat-preview-with-access-grants/packages/discussions/lib/Notifications.js#L63)